WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -0,0 +1,254 @@
+                    }, set: { value in
+                        self.fieldValues[dimensionType] = value
+                    }))
+                }
+            }
+            Toggle(isOn: $showSaveTemplate) {
+                Text(Localization.saveNewPackageTemplate)
+                    .font(.subheadline)
+            }
+            if showSaveTemplate {
+                Button(Localization.savePackageTemplate) {
+                    // save template
+                }
+                .buttonStyle(SecondaryButtonStyle())
+            }
+        }
+        .padding()
+    }
+
+    private var carrierPackageView: some View {
+        // TODO: just a placeholder
+        Text("carrier")
+    }
+
+    private var savedPackageView: some View {
+        // TODO: just a placeholder
+        Text("saved")
+    }
+}
+
+struct WooShippingAddPackageDimensionView: View {
+    enum DimensionType: CaseIterable {
+        case length, width, height
+        var name: String {
+            switch self {
+            case .length:
+                return Localization.length
+            case .width:
+                return Localization.width
+            case .height:
+                return Localization.height
+            }
+        }
+    }
+
+    let dimensionType: DimensionType
+    @Binding var fieldValue: String
+    @FocusState var fieldFocused: Bool
+
+    var body: some View {
+        VStack {
+            HStack {
+                Text(dimensionType.name)
+                    .font(.subheadline)
+                Spacer()
+            }
+            HStack {
+                TextField("", text: $fieldValue)
+                    .keyboardType(.decimalPad)
+                    .font(.body)
+                    .focused($fieldFocused)
+                Text("cm")
+                    .font(.subheadline)
+                    .foregroundStyle(.secondary)
+            }
+            .padding()
+            .roundedBorder(cornerRadius: 8,
+                           lineColor: fieldFocused ? Color(UIColor.wooCommercePurple(.shade60)) : Color(.separator),
+                           lineWidth: fieldFocused ? 2 : 1)
+        }
+        .frame(minHeight: 48)
+    }
+}
+
+#Preview {
+    WooShippingAddPackageView()
+}
+
+extension WooShippingAddPackageView {
+    enum Localization {
+        static let addPackage = NSLocalizedString("wooShipping.createLabel.addPackage.title",
+                                                  value: "Add Package",
+                                                  comment: "Title for the Add Package screen")
+        static let packageType = NSLocalizedString("wooShipping.createLabel.addPackage.packageType",
+                                                   value: "Package type",
+                                                   comment: "Info label for selecting package type")
+        static let cancel = NSLocalizedString("wooShipping.createLabel.addPackage.cancel",
```

Comment:
I see there is also "wooShipping.createLabel.cancelButton", I suppose our translation workflow will reuse the transations?

Reply:
Translations with different keys aren't automatically reused (in case the different context requires a different translation) but the translation memory tools in our workflow will suggest the translation for reuse. It feels like we probably don't need separate keys for strings like this that are used consistently in the UI, but for now my understanding is this is how we should be doing it.